rllib/evaluation/env_runner_v2.py

```diff
@@ -387,15 +387,6 @@ def run(self) -> Iterator[SampleBatchType]:
     def step(self) -> List[SampleBatchType]:
         """Samples training episodes by stepping through environments."""
 
-        # Before the very first poll (this will reset all vector sub-environments):
```

Comment:
I think this will break the `on_episode_created` callbacks. These need to fire before(!) the first reset happens. If we wait until after the initial `base_env.poll()`, then the resets on the individual sub-environments will have already happened.